src/plugins/condition/components/Condition.vue

```diff
@@ -34,33 +157,133 @@ export default {
         currentConditionIdentifier: {
             type: Object,
             required: true
+        },
+        conditionIndex: {
+            type: Number,
+            required: true
+        },
+        isEditing: {
+            type: Boolean,
+            required: true
+        },
+        telemetry: {
+            type: Array,
+            required: true,
+            default: () => []
         }
     },
     data() {
         return {
-            condition: this.condition
+            domainObject: this.domainObject,
+            currentCriteria: this.currentCriteria,
+            expanded: true,
+            trigger: 'all',
+            selectedOutputKey: '',
+            stringOutputField: false,
+            outputOptions: ['false', 'true', 'string']
         };
     },
-    destroyed() {
-        this.conditionClass.off('conditionResultUpdated', this.handleConditionResult.bind(this));
-        if (this.conditionClass && typeof this.conditionClass.destroy === 'function') {
-            this.conditionClass.destroy();
+    computed: {
+        initCap: function (string) {
+            return string.charAt(0).toUpperCase() + string.slice(1)
         }
     },
+    destroyed() {
+        this.destroy();
+    },
     mounted() {
-        this.openmct.objects.get(this.conditionIdentifier).then((obj => {
-            this.condition = obj;
-            this.conditionClass = new ConditionClass(this.condition, this.openmct);
-            this.conditionClass.on('conditionResultUpdated', this.handleConditionResult.bind(this));
+        this.openmct.objects.get(this.conditionIdentifier).then((domainObject => {
+            this.domainObject = domainObject;
+            this.initialize();
         }));
     },
+    updated() {
+        //validate telemetry exists, update criteria as needed
+        this.persist();
```

Comment:
This is not efficient. We need to remove this `persist()` call on updated and call it only when we want things to persist.

Reply:
done